socorro/external/es/supersearch.py

```diff
@@ -437,27 +441,32 @@ def get(self, **kwargs):
 
             search.aggs.bucket('signature', sig_bucket)
 
-        # Create date histograms.
-        if params.get('_histogram.date'):
-            date_bucket = A(
-                'date_histogram',
-                field=self.get_field_name('date'),
-                interval=histogram_interval_date,
-            )
-            for param in params['_histogram.date']:
-                for value in param.value:
-                    if not value:
-                        continue
-
-                    field_name = self.get_field_name(value)
-                    val_bucket = A(
-                        'terms',
-                        field=field_name,
-                        size=facets_size,
-                    )
-                    date_bucket.bucket(value, val_bucket)
-
-            search.aggs.bucket('histogram_date', date_bucket)
+        # Create histograms.
+        for f in self.histogram_fields:
+            if params.get('_histogram.%s' % f):
+                histogram_type = (
+                    self.all_fields[f]['query_type'] == 'date'
+                    and 'date_histogram' or 'histogram'
+                )
+                date_bucket = A(
+                    histogram_type,
+                    field=self.get_field_name(f),
+                    interval=histogram_intervals[f],
+                )
+                for param in params['_histogram.%s' % f]:
+                    for value in param.value:
+                        if not value:
+                            continue
+
+                        field_name = self.get_field_name(value)
+                        val_bucket = A(
+                            'terms',
+                            field=field_name,
+                            size=facets_size,
+                        )
+                        date_bucket.bucket(value, val_bucket)
```

Comment:
I understood this briefly when you walked me through this in Whistler when it was just the regular histogram or whatever it was then. I understood it then. Now it has escaped my limited mind :)
But I can see that this block above is just that stuff but done in a loop. I hope it's correct :)